packages/kbn-unified-data-table/src/components/row_height_settings.tsx

```diff
@@ -67,14 +72,14 @@ export function RowHeightSettings({
             legend={label}
             buttonSize="compressed"
             options={rowHeightModeOptions}
-            idSelected={`${idPrefix}${rowHeight ?? 'single'}`}
+            idSelected={`${idPrefix}${rowHeight ?? RowHeight.Single}`}
```

Comment:
We could also replace it in `id`s and `data-test-subj`s above in `rowHeightModeOptions` to make sure that a right value gets into `onChangeRowHeight(newMode)`.